salt/iiif/init.sls

```diff
@@ -0,0 +1,221 @@
+        - require:
+            - loris-user
+            - mount-external-volume
+
+loris-cache-resolver:
+    file.directory:
+        - name: {{ pillar.iiif.loris.storage }}/cache-resolver
+        - user: {{ loris_user }}
+        - group: {{ loris_user }}
+        - makedirs: True
+        - require:
+            - loris-user
+            - mount-external-volume
+
+# empty folder that can be synced over the caches to clean them
+loris-cache-blank:
+    file.directory:
+        - name: {{ pillar.iiif.loris.storage }}/blank
+        - user: {{ loris_user }}
+        - group: {{ loris_user }}
+        - dir_mode: 755
+        - makedirs: True
+        - require:
+            - loris-user
+            - mount-external-volume
+
+
+# Docker needs to bind certain configuration files between host and container
+# this is where you put them
+loris-dir:
+    file.directory:
+        - user: {{ pillar.elife.deploy_user.username }}
+        - name: /opt/loris
+
+loris-config:
+    file.managed:
+        - user: {{ pillar.elife.deploy_user.username }}
+        - name: /opt/loris/loris2.conf
+        - source: salt://iiif/config/opt-loris-loris2.conf
+        - template: jinja
+        - makedirs: True
+        - require:
+            - loris-dir
+
+# required by newrelic-python.sls
+loris-newrelic-venv:
+    cmd.run:
+        - cwd: /opt/loris
+        - runas: {{ pillar.elife.deploy_user.username }}
+        - name: |
+            python3 -m venv venv
+            venv/bin/pip install newrelic==5.8.0.136
+        - unless:
+            - test -d /opt/loris/venv
+            
+# required by newrelic-python.sls because it's using builder-private and not the formula's pillar
+# remove once builder-private changes are in and the service is removed
+loris-uwsgi-ready:
+    service.running:
+        - name: nginx # could be anything that should be enabled by default
+
+# required by newrelic-python.sls because it's using builder-private and not the formula's pillar
+# remove once builder-private changes are in
+loris-setup:
+    cmd.run:
+        - name: "echo dummy state"
+        - require:
+            - loris-config
+            - loris-newrelic-venv
+            - loris-uwsgi-ready
+
+
+# newrelic-python runs about here
+
+
+loris-uwsgi-config:
+    file.managed:
+        # systemd service file expects to find uwsgi.ini in app folder
+        # see builder-base.uwsgi
+        - name: /opt/loris/uwsgi.ini
+        - source: salt://iiif/config/opt-loris-uwsgi.ini
+        - template: jinja
+        - require:
+            - loris-dir
+
+loris-wsgi-entry-point:
+    file.managed:
+        - name: /opt/loris/loris2.wsgi
+        - source: salt://iiif/config/opt-loris-loris2.wsgi
+        - require:
+            - loris-dir
+
+# `docker_container.running` state:
+# - https://docs.saltstack.com/en/latest/ref/states/all/salt.states.docker_container.html#salt.states.docker_container.running
+
+
+{% if pillar.elife.env == "dev" %}
+
+build-loris:
```

Comment:
what to do about this? extremely useful while debugging but very specific to my own vagrant setup

perhaps add a 'onlyif: /vagrant/loris-docker' exists type thing